subsys/bluetooth/audio/ascs.c

```diff
@@ -387,52 +396,92 @@ static void ase_set_state_releasing(struct bt_ascs_ase *ase)
 
 static void state_transition_work_handler(struct k_work *work)
 {
-	struct bt_ascs_ase *ase = CONTAINER_OF(work, struct bt_ascs_ase, state_transition_work);
+	struct k_work_delayable *d_work = k_work_delayable_from_work(work);
+	struct bt_ascs_ase *ase = CONTAINER_OF(d_work, struct bt_ascs_ase, state_transition_work);
 	const enum bt_bap_ep_state old_state = ascs_ep_get_state(&ase->ep);
 	const enum bt_bap_ep_state new_state = ase->state_pending;
 	struct bt_bap_stream *stream = ase->ep.stream;
-	struct bt_bap_ep *ep = &ase->ep;
+	int err;
 
 	__ASSERT_NO_MSG(stream != NULL);
 
-	/* We left the streaming state, let the upper layers know that the stream is stopped */
-	if (old_state != new_state && old_state == BT_BAP_EP_STATE_STREAMING) {
-		struct bt_bap_stream_ops *ops = stream->ops;
-		uint8_t reason = ep->reason;
+	ase->ep.status.state = new_state;
+
+	/* Notify ASE state */
+	if (ase->conn != NULL) {
+		struct bt_conn_info info;
+
+		err = ase_state_notify(ase);
+		if (err == -ENOMEM) {
+			uint32_t retry_delay_ms;
+
+			/* Revert back to old state */
+			ase->ep.status.state = old_state;
```

Comment:
Not specific for notifications, but how do we handle this if a client sends another request while we already have a state pending? Do we auto-reject any additional requests, or do we overwrite the pending state?